cloudpathlib/cloudpath.py

```diff
@@ -594,14 +594,15 @@ def download_to(self, destination: Union[str, os.PathLike]) -> Path:
                 destination = destination / self.name
             return self.client._download_file(self, destination)
         else:
-            destination.mkdir(exist_ok=True)
+            destination.mkdir(parents=True, exist_ok=True)
             for f in self.iterdir():
                 rel = str(self)
                 if not rel.endswith("/"):
                     rel = rel + "/"
 
                 rel_dest = str(f)[len(rel) :]
-                f.download_to(destination / rel_dest)
+                if rel_dest != "":
```

Comment:
The root cause of this is that `iterdir` improperly yields itself (issue #15), which we have a fix for in #202 that should be merged and released soon.  

This makes the check here redundant. I don't immediately have a strong leaning on whether having the redundant check would be good or bad. Either way, it would also be useful is to add a test for this case.